### PR TITLE
Feature/tao 3496 app controller

### PR DESCRIPTION
**Pull request by jsconan**
Related to: https://oat-sa.atlassian.net/browse/TAO-3496

Add application controller:
- instantiate the historyRouter
- catch routable links
- instantiate a logger
- provide an error handler

Improve the historyRouter:
- use promises
- fix history state on launch
- add actions: `redirect()`, `forward()`, `replace()`

### CODE REVIEW

File: views/js/core/app.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Jean-Sébastien Conan <jean-sebastien@taotesting.com>
+ */
+define([
```
Comment (krampstudio) [inline views/js/core/app.js]: We should rename it to `core/appController.js` or may be move it to `controllers/app`. It depends if it is loaded directly by the bootstrap or not.

Comment (jsconan) [inline views/js/core/app.js]: I renamed it to `core/appController` as it is more a companion module than a bootstrap: there is not `start` method.

File: views/js/controller/app.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+     * Shared router that will manage the page for each controller
+     * @type {historyRouter}
+     */
+    var historyRouter;
+
+    /**
+     * Creates a logger for the app
+     * @type {logger}
+     */
+    var appLogger = loggerFactory('application');
```
Comment (krampstudio) [inline views/js/controller/app.js]: by convention the logger takes the module name

Comment (jsconan) [inline views/js/controller/app.js]: Oops, I did know that. I will fix it.

Comment (jsconan) [inline views/js/controller/app.js]: Fixed!

**Pull request merged**